<?xml version="1.0" encoding="utf-8"?>
<pysystest type="auto">

    <description>
        <title>Error: revert on non-view function via ethers and web3js</title>
        <purpose><![CDATA[
In the test we check the web3js and ethers library handling of a transaction that fails due to a revert, specifically
we check that the libraries don't abstract over doing a call on the function tx in order to get the revert reason, and
therefore that it is the developers responsibility to get it.
]]>
        </purpose>
    </description>

    <classification>
        <groups inherit="true">
            <group>ethers</group>
            <group>web3js</group>
        </groups>
        <modes inherit="true">
            <mode>ten.sepolia</mode>
            <mode>ten.uat</mode>
            <mode>ten.dev</mode>
            <mode>ten.local</mode>
            <mode>arbitrum.sepolia</mode>
            <mode>ganache</mode>
            <mode>sepolia</mode>
        </modes>
    </classification>

    <data>
        <class name="PySysTest" module="run"/>
    </data>

    <traceability>
        <requirements>
            <requirement id=""/>
        </requirements>
    </traceability>
</pysystest>
